gatsby/content/blog/2022/10/2022-10-04-faster-joins.mdx

```diff
@@ -0,0 +1,45 @@
+---
+date: '2022-10-04'
+title: Testing faster remote room joins
+categories:
+  - General
+author: Richard van der Hoff
+---
+We’re very excited to announce that, as of Synapse 1.69, "faster remote room joins" are ready for community testing.
+
+There are a number of caveats, which I'll come to, but first: this is an important step in a project which we've been working on for 9 months. Most people who use Matrix will be familiar with the pain of joining a large room over federation: typically you are just faced with a spinner, which is eventually replaced by a cryptic error. If you're lucky, the room eventually pops up in your room list of its own accord. The whole experience is [one of the longest-standing open issues in Synapse](https://github.com/matrix-org/synapse/issues/1211).
+
+The "faster joins" project set out to change all this. Briefly, the reason the experience is so poor today is that there is a *lot* of information required to join a large room, which is slow for the "resident" server to generate, and even slower for the "joining" server to validate and store. The key insight is that we don't really need the full membership list of a room to get started: rather, we can present the user with the basic information about the room (name, topic, space hierarchy, etc), and then populate the full details in the background. The upshot is that, whereas it used to take upwards of 12 minutes to join Matrix HQ (even assuming that your server didn't fall over in the meantime), this is now down to about 30 seconds (and we're confident that we can reduce this even further).
+
+It sounds simple enough, but in reality this has meant a lot of work on Synapse. The main problem was that huge amounts of the Synapse codebase assumed that we either had all the state for a room, or none of it. That assumption no longer holds, which meant lots of changes to the code to handle "partial state". Along the way we've also had to make some other improvements: for example, we [found and fixed](https://github.com/matrix-org/synapse/security/advisories/GHSA-jhjh-776m-4765) some long-standing bugs in event authorisation, and implemented [cancellation](https://github.com/matrix-org/synapse/issues/3528) for requests where the client disconnects before we finish handling the request. For an idea of the things being changed at the Matrix protocol level, take a look at [MSC3902](https://github.com/matrix-org/matrix-spec-proposals/pull/3902).
+
+The upshot of all that, is that instead of seeing a blank room with a spinner (or an error message), you can now see the basic information about the room, as well as receive incoming messages, within a few seconds of trying to join the room:
+
+![Element screenshot](/blog/img/2022-10-04-joining-hq.png)
```

Comment:
I'm not sure this screenshot is as compelling as it might be. I was hoping that there might be some live traffic in HQ, but of course for once it was silent when I had the test rig running.

Any thoughts?

Reply:
It seems to convey the opposite message of what it's meant to. We can either set-up a fresh demo room, or have a silly conversation with a few folks in MatrixHQ itself while you're attempting a join?

Reply:
I'm just going to remove the screenshot for now.